manual/case_sensitivity/README.md

```diff
@@ -0,0 +1,129 @@
+## Case sensitivity
+
+### In Cassandra
+
+Cassandra identifiers, such as keyspace, table and column names, are case-insensitive by default.
+For example, if you create the following table:
+
+```
+cqlsh> CREATE TABLE test.FooBar(k int PRIMARY KEY);
+```
+
+Cassandra actually stores the table name as lower-case:
+
+```
+cqlsh> SELECT table_name FROM system_schema.tables WHERE keyspace_name = 'test';
+
+ table_name
+------------
+     foobar
+```
+
+And you can use whatever case you want in your queries:
+
+```
+cqlsh> SELECT * FROM test.FooBar;
+cqlsh> SELECT * FROM test.foobar;
+cqlsh> SELECT * FROM test.FoObAr;
+```
+
+However, if you enclose an identifier in double quotes, it becomes case-sensitive:
+
+```
+cqlsh> CREATE TABLE test."FooBar"(k int PRIMARY KEY);
+cqlsh> SELECT table_name FROM system_schema.tables WHERE keyspace_name = 'test';
+
+ table_name
+------------
+     FooBar
+```
+
+You now have to use the exact, quoted form in your queries:
+
+```
+cqlsh> SELECT * FROM test."FooBar";
+```
+
+If you forget to quote, or use the wrong case, you'll get an error: 
+
+```
+cqlsh> SELECT * FROM test."FOOBAR";
+InvalidRequest: Error from server: code=2200 [Invalid query] message="table FOOBAR does not exist"
+
+cqlsh> SELECT * FROM test.Foobar;
+InvalidRequest: Error from server: code=2200 [Invalid query] message="table foobar does not exist"
+```
+
+### In the driver
+
+When we manipulate identifiers, we use the following definitions:
+
+* **CQL form**: how you would type it in a CQL query. In other words, case-sensitive if it's quoted,
+  case-insensitive otherwise; 
+* **internal form**: how it is stored in system tables. In other words, never quoted and always in
+  its exact case.
+
+In previous driver versions, identifiers were represented as raw strings. The problem is that this
+does not capture the form; when a method processed an identifier, it always had to know where it
+came from and what form it was in, and possibly convert it. This led a lot of internal complexity,
+and recurring bugs.
+
+To address this issue, driver 4+ uses a wrapper: [CqlIdentifier]. Its API methods are always
+explicit about the form:
+
+```java
+CqlIdentifier caseInsensitiveId = CqlIdentifier.fromCql("FooBar");
+System.out.println(caseInsensitiveId.asInternal());             // foobar
+System.out.println(caseInsensitiveId.asCql(/*pretty=*/ false)); // "foobar"
+System.out.println(caseInsensitiveId.asCql(true));              // foobar
+
+// Double-quotes need to be escaped inside Java strings
+CqlIdentifier caseSensitiveId = CqlIdentifier.fromCql("\"FooBar\"");
+System.out.println(caseSensitiveId.asInternal());               // FooBar
+System.out.println(caseSensitiveId.asCql(true));                // "FooBar"
+System.out.println(caseSensitiveId.asCql(false));               // "FooBar"
+```
+
+*Side note: as shown above, `asCql` has a pretty-printing option that omits the quotes if possible.
```

Comment:
When I initially read this it was sort of unclear what was meant by 'if possible', I think it would be better phrase it as 'if they are not necessary'